HoneyDo.Web/ClientApp/src/pages/Tasks.tsx

```diff
@@ -85,6 +85,21 @@ class Tasks extends Component<TasksProps, TasksState> {
     this.updateExistingTask(updatedTask);
   };
 
+  handleDelete = async (task: Task) => {
+    const { deleteTask }: TaskContextData = this.context;
+    const isDeleted = await deleteTask(task);
+    if (!isDeleted) {
+      return;
+    }
+    const { tasks } = this.state;
+    const taskIndex = tasks.findIndex(tsk => tsk.id === task.id);
+    const updateTasks: Task[] = [
+      ...tasks.slice(0, taskIndex),
+      ...tasks.slice(taskIndex + 1)
+    ];
+    this.setState({ tasks: updateTasks, editTask: undefined });
```

Comment:
Should `editTask` be null or false instead of undefined? I know undefined returns falsely, but this feels weird.

Reply:
editTask is the task being edited, so I think it should be a task or nothing (undefined).

Reply:
Ah okay now I get the undefined. Why undefined over null? Are null exceptions possible if you do that instead of `null`?

Reply:
`null` could work too, I defined the state as `editTask?: Task`, the `?` means `editTask` is optional, and to set it as empty or blank typescript wants us to use `undefined`.  If we wanted to use null instead you'd have to define it like this `editTask: Task | null` or `editTask?: Task | null` I think we might as well just use `undefined`... typescript thing.